src/rateEngine/LoadProfileScaler.ts

```diff
@@ -24,58 +34,69 @@ class LoadProfileScaler {
     return this.to(scaler);
   }
 
-  toAverageMonthlyBill(
-    amount: number,
-    rate: RateInterface
-  ): LoadProfile {
-    // Need this as a percent for better resolution within the algorithm
-    const scalerAsPercent = 100;
-    const fnParams = [scalerAsPercent, rate, this];
+  toAverageMonthlyBill(amount: number, rate: RateInterface, goalSeekParams: GoalSeekArgs = {}): LoadProfile {
+    const magnitude = Math.max(Math.floor(Math.log10(Math.abs(amount))), 0);
+
+    const magnitudeScaler = 10 ** magnitude;
+    const initialScalerGuess = magnitudeScaler;
+    const fnParams = [initialScalerGuess, rate, this, magnitude];
 
     try {
-      const finalScalerAsPercent = goalSeek({
+      const finalScaler = goalSeek({
         fn: this.scaledMonthlyCost,
         fnParams,
         percentTolerance: 0.1,
         maxIterations: 1000,
-        maxStep: 50,
+        maxStep: 0.5 * magnitudeScaler,
         goal: amount,
-        independentVariableIdx: 0
+        independentVariableIdx: 0,
+        ...goalSeekParams,
       });
-      return this.to(finalScalerAsPercent / 100);
+
+      const scalerAsDecimal = finalScaler / magnitudeScaler;
+      return this.to(scalerAsDecimal);
     } catch (e) {
-      throw(e);
+      throw e;
     }
   }
 
   toMonthlyKwh(monthlyKwh: Array<number>): LoadProfile {
     if (monthlyKwh.length !== 12) {
-      throw('monthlyKwh must be an array of 12 numbers');
+      throw 'monthlyKwh must be an array of 12 numbers';
     }
     const scalersByMonth = this.loadProfile.sumByMonth().map((kwh, idx) => {
       return monthlyKwh[idx] / kwh;
     });
-    const scaledLoad = this.loadProfile.expanded().map(loadHour => {
+    const scaledLoad = this.loadProfile.expanded().map((loadHour) => {
       return {
         ...loadHour,
         load: loadHour.load * scalersByMonth[loadHour.month],
       };
     });
-    return new LoadProfile(scaledLoad, {year: this.loadProfile.year});
+    return new LoadProfile(scaledLoad, { year: this.loadProfile.year });
   }
 
   private scaledMonthlyCost(
-    scalerAsPercent: number,
+    scaler: number,
     rate: RateInterface,
     context: LoadProfileScaler,
+    magnitude: number,
   ): number {
-    const scaledLoadProfile = context.to(scalerAsPercent / 100);
+    const scaledLoadProfile = context.to(scaler / 10 ** magnitude);
     const rateCalculator = new RateCalculator({
       ...rate,
       loadProfile: scaledLoadProfile,
     });
-    return rateCalculator.annualCost() / 12;
-  };
-};
+
+    const currentMonthlyCost = rateCalculator.annualCost() / 12;
+
+    if (context.debug) {
+      console.log('current scaler is:', scaler);
+      console.log('current monthlyCost is:', currentMonthlyCost);
+    }
```

Comment:
This is very helpful to see if you have a goal seek that won't converge. Exposing this means we can debug what's going on with a non-converging scaling operation in the application using the package rather than needing to dig into the engine internals immediately